driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2043,8 +2117,16 @@ public void run() {
                         schemaInAgreement = ControlConnection.waitForSchemaAgreement(connection, Manager.this);
                         if (!schemaInAgreement)
                             logger.warn("No schema agreement from live replicas after {} s. The schema may not be up to date on some nodes.", configuration.getProtocolOptions().getMaxSchemaAgreementWaitSeconds());
-                        if (refreshSchema)
-                            ControlConnection.refreshSchema(connection, keyspace, table, Manager.this, false);
+                        if (refreshSchema) {
+                            submitSchemaRefresh(keyspace, table);
+                            // force synchronous schema refresh
+                            Uninterruptibles.getUninterruptibly(schemaRefreshRequestDebouncer.deliverEvents());
```

Comment:
Actually the previous version performed the refresh with the _current_ connection (the one that just executed the schema-altering statement), while the debouncer will always do it with the control connection.
The problem is that the control host might not yet be up to date with the schema changes, so the metadata is still stale when the statement completes.
I'm not quite sure how to solve that yet.